modules/systemd-journal/journald-subsystem.h

```diff
@@ -65,25 +65,44 @@ union sd_id128
 
 typedef struct sd_journal sd_journal;
 
-extern int (*sd_journal_open)(sd_journal **ret, int flags);
+/*
+ * attribute((visibility("hidden")) is needed to avoid these variables to be
+ * overridden with symbols of the same name from libsystemd.so.
+ *
+ * It may happen that libsyslog-ng.so itself is linked against libsystemd.so
+ * (due to service management being used).  In this case these names would
+ * map to actual functions in libsystemd.so, resulting in a nasty crash
+ * whenever we try to use these as pointers to functions.
+ *
+ * By using "hidden" visibility, we tell the linker that these symbols are
+ * internal to the current .so, meaning that any code that references these
+ * from this module will happily use these as pointers while other modules
+ * of syslog-ng can still use them as functions.
+ *
+ * See https://github.com/syslog-ng/syslog-ng/issues/4300 for more details.
+ */
+
+#define VISIBILITY_HIDDEN __attribute__((visibility("hidden")))
```

Comment:
this is what I was looking for after I realized that **g_module_symbol** finds the given symbol in our plugin first not in the module passed as its first parameter that is, in my opinion, a glib bug